cdap-security/src/main/java/co/cask/cdap/security/server/JAASLoginService.java

```diff
@@ -309,4 +331,53 @@ public void logout(UserIdentity user) {
     }
   }
 
+  private String getLDAPURL(Hashtable map){
+    StringBuilder sb = new StringBuilder();
+    if(map.containsKey("useLdaps") && map.get("useLdaps").toString().equalsIgnoreCase("true")){
+      sb.append("ldaps");
+    } else{
+      sb.append("ldap");
+    }
+    sb.append("://");
+    sb.append(map.get("hostname").toString()).append(":").append(map.get("port"));
+
+    return sb.toString();
+  }
+
+  private String convertToCamelCase(String userName){
+    return org.apache.commons.lang3.text.WordUtils.capitalizeFully(userName, ldap_CamelCaseConversionDelimiterArr);
+  }
+
+  private String getUserNameFromLDAP(String userName){
+    try{
+      Hashtable map = new Hashtable(configuration.getAppConfigurationEntry("co.cask.cdap.security.server.LDAPLoginModule")[0].getOptions());
+      map.put(Context.INITIAL_CONTEXT_FACTORY, map.get("contextFactory").toString());
+      map.put(Context.SECURITY_AUTHENTICATION, map.get("authenticationMethod").toString());
+      map.put(Context.PROVIDER_URL, getLDAPURL(map));
+
+      LdapContext ctx = new InitialLdapContext(map, null);
+
+      SearchControls constraints = new SearchControls();
+      constraints.setSearchScope(SearchControls.SUBTREE_SCOPE);
+
+      String ldapHandlerBinding = ctx.getEnvironment().get("userBaseDn").toString();
+
+      String filterExpr = "(" + ctx.getEnvironment().get("userRdnAttribute").toString() + "=" + userName + ")";
+
+      NamingEnumeration answer = ctx.search(ldapHandlerBinding, filterExpr, constraints);
+
+      if (answer.hasMore()) {
+        Attributes attrs = ((SearchResult) answer.next()).getAttributes();
+        userName = (String)attrs.get(ctx.getEnvironment().get("userRdnAttribute").toString()).get();
+        LOG.debug("applying case adjustment using data from LDAP, updated username:: " + userName);
+      } else{
+        LOG.warn("case adjustment not done for username, user not found in LDAP:: " + userName);
+      }
+
+    }catch (Exception e){
+      LOG.warn("Exception occurred while get user details from LDAP, not able to apply case adjustment " + e.getMessage());
+      e.printStackTrace();
+    }
+    return userName;
+  }
 }
```

Comment:
These functions are LDAP specific, should go into LDAP specific implementation class.

Reply:
ContextFactory is being set to only one value. CDAP supports both LDAP and LDAPS. Please check if this works with LDAPS

Reply:
> ContextFactory is being set to only one value. CDAP supports both LDAP and LDAPS. Please check if this works with LDAPS

created https://guavus-jira.atlassian.net/browse/RAFD-2974 to track it